### PR TITLE
ADR 006 - consistent API for components that fill the parent width

### PR DESCRIPTION
**Pull request by mperrotti**
Proposes a consistent API for making a component fill its parent width

### CODE REVIEW

File: contributor-docs/adrs/adr-006-component-fill-width-api.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# ADR 006: Use consistent API for making a component fill a parent's width
+
+## Status
+
+Proposed
+
+## Context
+
+By default, components like [Button](https://primer.style/react/Button) and [TextInput](https://primer.style/react/TextInput) are layed out inline, and are only do not fill the width of their parent. However, there are some layouts where these components fill the width of their parent for better ergonomics or to raise its level of hierarchy on the page.
```
Comment (langermank) [inline contributor-docs/adrs/adr-006-component-fill-width-api.md:9]:
```suggestion
By default, components like [Button](https://primer.style/react/Button) and [TextInput](https://primer.style/react/TextInput) are displayed inline, and do not expand to fill the width of their parent. However, there are some layouts where these components fill the width of their parent for better ergonomics or to raise its level of hierarchy on the page.
```

**Review by langermank — APPROVED** [review]
Love this! Aligns perfectly with the [Button API proposal](https://github.com/github/design-infrastructure/discussions/2310)

Thank you!

**Review by colebemis — APPROVED** [review]
Thanks for writing this, @mperrotti! Looks good to me 💖 

Could you move this ADR to [github/primer](https://github.com/github/primer/tree/main/adrs), since this decision impacts ViewComponent APIs as well?

**Comment by mperrotti** [comment]
Moved to https://github.com/github/primer/pull/1017

**Pull request closed without merging**